### ISSUE TITLE
Products Block: Layout Options

### ISSUE THREAD

**Issue by douglas-johnson** [thoughtis/cata-blocks#50]
### Background

We are interested in more layout options for the products block. Currently we have a single grid-like layout.

Designs are not finalized, but we're looking into a "reel" type layout that displays a single row of products and allows scrolling when the content of that row is wider than its container.

### Questions

How do we want to apply this layout change?

https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/block-controls-toolbar-and-sidebar/

#### Thought Catalog example

The Thought Catalog REST API block has two layout options. We apply them using a select input in the sidebar panel.

All styles are handled by us and the layouts are coded twice, in React and PHP, and the resulting HTML is saved to the post.

https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/inspector-controls/README.md

<img width="1365" alt="Screen Shot 2022-11-10 at 1 47 16 PM" src="https://user-images.githubusercontent.com/226381/201180719-5ba4a760-e9ca-489c-aa46-6c87b8c65cd2.png">

#### WordPress core example

The core query loop block uses the formatting toolbar for basic layout options like grid and list.

Further options are handled through the post template block. I do not totally understand how it works. It saves the query parameters and the options for each child of the post template block. So, it's possible to utilize things like font-size controls.

https://developer.wordpress.org/block-editor/reference-guides/slotfills/plugin-block-settings-menu-item/

https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query/edit/query-toolbar.js
https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-template

<img width="1202" alt="Screen Shot 2022-11-10 at 1 49 52 PM" src="https://user-images.githubusercontent.com/226381/201181144-48f71041-77c0-482c-a524-757f3a1e16a6.png">

**Comment by douglas-johnson** [comment]
Additional options to layout updates where only CSS needs to change, not the HTML output.

https://developer.wordpress.org/block-editor/reference-guides/block-api/block-styles/
https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/

**Comment by douglas-johnson** [comment]
The most recent time I added a style was the streaming button. Here's the related pull request. I wish I had written a better PR message. I would like to know exactly why I wrote styles in our CSS file as well as inline CSS. I think it was for the purpose of visualizing the streaming button in the editor.

https://github.com/thoughtis/cata/pull/109/files

**Comment by jessica-townsend** [comment]
I experimented with the [block styles](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-styles/) option which seemed to work very well. I feel that this would be the best option as long as the final design can be reasonably implemented without changing any markup.

Here's the branch where I made the changes:
https://github.com/thoughtis/cata-blocks/tree/issue%2350/add-new-product-block-style

Here's a screenshot of the style selector in the editor:

<img width="279" alt="Screen Shot 2022-11-11 at 10 43 36 AM" src="https://user-images.githubusercontent.com/70295707/201381494-7fc07aa2-bcbc-4231-9489-cf4399ee4119.png">


Here's the resulting class name for the block wrapper:
```
<div class="wp-block-cata-products is-style-reel alignwide">...</div>
```

**Comment by douglas-johnson** [comment]
Editor styles from the reel block on Thought Catalog
https://github.com/wpcomvip/thought-catalog/blob/master/plugins/ought-blocks/blocks/reel/src/editor.scss

Shared styles between editor and end-user that might interfere with `is-style-reel` 
https://github.com/thoughtis/cata-blocks/blob/main/blocks/products/src/style.scss

I will work on possibly adding `is-style-grid` as the default so there is less to undo in your work

**Comment by jessica-townsend** [comment]
Using `get_block_wrapper_attributes` mostly worked for me. The class `alignwide` had to be added as an extra attribute, but I feel that this solution is still a lot simpler than having a custom function to add class names.

New render function here:
https://github.com/thoughtis/cata-blocks/blob/4126f30678253ff41069d39b99c680bbec8d681c/blocks/products/includes/renderer.php#L17-L36

@douglas-johnson let me know if you think the `alignwide` class could be handled differently.

**Comment by douglas-johnson** [comment]
@jessica-townsend I tried this, and I saw that if we add add even the most basic save component to our block, `alignwide` is incorporated into the classes for `get_block_wrapper_attributes`.

Adding the save component is a lightweight breaking change.
- It will cause the editor to throw an error and ask the user to "attempt block recovery", which should work.
- Existing blocks will lose their `alignwide` based on the updated render callback, until the block is opened in the editor, recovered and saved.

Cool if I break this out into a separate issue so we don't have to give everything a minor break AND ship the style update at the same time?
